scripts/monorepo/align-package-versions.js

```diff
@@ -0,0 +1,146 @@
+    consumerPackageAbsolutePath,
+    'package.json',
+  );
+  const consumerPackageManifest = readJSONFile(consumerPackageManifestPath);
+
+  const dependencyVersion =
+    consumerPackageManifest.dependencies?.[updatedPackageName];
+
+  if (dependencyVersion && dependencyVersion !== '*') {
+    const updatedDependencyVersion = dependencyVersion.startsWith('^')
+      ? `^${updatedPackageVersion}`
+      : updatedPackageVersion;
+
+    if (updatedDependencyVersion !== dependencyVersion) {
+      console.log(
+        `\uD83D\uDCA1 ${consumerPackageManifest.name} was updated: now using version ${updatedPackageVersion} of ${updatedPackageName}`,
+      );
+
+      const updatedPackageManifest = {
+        ...consumerPackageManifest,
+        dependencies: {
+          ...consumerPackageManifest.dependencies,
+          [updatedPackageName]: updatedDependencyVersion,
+        },
+      };
+
+      writeFileSync(
+        consumerPackageManifestPath,
+        JSON.stringify(updatedPackageManifest, null, 2) + '\n',
+        'utf-8',
+      );
+    }
+  }
+
+  const devDependencyVersion =
+    consumerPackageManifest.devDependencies?.[updatedPackageName];
+
+  if (devDependencyVersion && devDependencyVersion !== '*') {
+    const updatedDependencyVersion = devDependencyVersion.startsWith('^')
+      ? `^${updatedPackageVersion}`
+      : updatedPackageVersion;
+
+    if (updatedDependencyVersion !== devDependencyVersion) {
+      console.log(
+        `\uD83D\uDCA1 ${consumerPackageManifest.name} was updated: now using version ${updatedPackageVersion} of ${updatedPackageName}`,
+      );
+
+      const updatedPackageManifest = {
+        ...consumerPackageManifest,
+        devDependencies: {
+          ...consumerPackageManifest.devDependencies,
+          [updatedPackageName]: updatedDependencyVersion,
+        },
+      };
+
+      writeFileSync(
+        consumerPackageManifestPath,
+        JSON.stringify(updatedPackageManifest, null, 2) + '\n',
+        'utf-8',
+      );
+    }
+  }
+};
+
+const alignPackageVersions = () => {
+  if (checkForGitChanges()) {
+    console.log(
+      '\u274c Found uncommitted changes. Please commit or stash them before running this script',
+    );
+
+    process.exit(1);
+  }
+
+  forEachPackage((packageAbsolutePath, _, packageManifest) => {
+    checkIfShouldUpdateDependencyPackageVersion(
+      ROOT_LOCATION,
+      packageManifest.name,
+      packageManifest.version,
+    );
+
+    checkIfShouldUpdateDependencyPackageVersion(
+      TEMPLATE_LOCATION,
+      packageManifest.name,
+      packageManifest.version,
+    );
+
+    checkIfShouldUpdateDependencyPackageVersion(
+      REPO_CONFIG_LOCATION,
+      packageManifest.name,
+      packageManifest.version,
+    );
+
+    forEachPackage(pathToPackage =>
+      checkIfShouldUpdateDependencyPackageVersion(
+        pathToPackage,
+        packageManifest.name,
+        packageManifest.version,
+      ),
+    );
+  });
```

Comment:
not a super big fan of this double forEach which feels a bit like brute forcing checking everything against everything but honestly hey at least we'll make sure that everything is correctly aligned!

And I think this will work out of the box with -stable branches too so 👍